pmd-kotlin/src/main/java/net/sourceforge/pmd/lang/kotlin/AbstractKotlinRule.java

```diff
@@ -0,0 +1,19 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.kotlin;
+
+import net.sourceforge.pmd.RuleContext;
+import net.sourceforge.pmd.lang.ast.AstVisitor;
+import net.sourceforge.pmd.lang.ast.impl.antlr4.AntlrBaseRule;
+
+public abstract class AbstractKotlinRule extends AntlrBaseRule {
+
+    protected AbstractKotlinRule() {
+        // inheritance constructor
+    }
+
+    @Override
+    public abstract AstVisitor<RuleContext, ?> buildVisitor();
```

Comment:
```suggestion
    public abstract KotlinVisitor<RuleContext, Void> buildVisitor();
```

Reply:
Where is the KotlinVisitor? We copied this from AbstractSwiftRule: there is no SwiftVisitor also?

Reply:
They should be generated by Antlr, aren't they?

Reply:
Yes, they are generated.